code/go/0chain.net/chaincore/node/n2n_send.go

```diff
@@ -296,13 +303,13 @@ func SendEntityHandler(uri string, options *SendOptions) EntitySendHandler {
 				selfNode.SetLastActiveTime(ts)
 				selfNode.InduceDelay(receiver)
 
-				cctx, cancel := context.WithTimeout(ctx, timeout)
-				defer cancel()
+				cctx, cancel = context.WithTimeout(ctx, timeout)
```

Comment:
do you think it is better to move `cctx, cancel = context.WithTimeout(ctx, timeout)` to line 297?

Reply:
No, actually we moved it into the `func()` block because the timeout context starts to count down immediately when it is created, we don't want the http request be canceled early due to the time waste on other processes such as `receiver.Grab()`.

Reply:
yes. I see. because `defer cancel()` was in `func()`. 
```suggestion

// move here, it is easier to understanding than inside `func()`
cctx, cancel := context.WithTimeout(ctx, timeout)

func() {
  receiver.Grab()
  defer receiver.Release()
  selfNode = Self.Underlying()
  selfNode.SetLastActiveTime(ts)
  selfNode.InduceDelay(receiver)

  req = req.WithContext(cctx)
  resp, err = httpClient.Do(req)
}()

	
defer cancel()
```

Reply:
Moving it outside of the `func()` block would waste the context time on `receiver.Grab()`, especially when we have lot of nodes, part of the sending requests will be canceled before we send http requests. Anyway, I'll refactor it to make it easier to read.